### PR TITLE
defining arc api

### PR DESCRIPTION
**Pull request by ranasaria**
This PR defines an arc API object returned by the arc extension. This is consumed in resource-deployment and quick test there verifies that all plumbing is working correctly. 

This PR has no direct functional change to the product and its behavior yet but sets up the plumbing needed to use arc extension objects through this newly created API in other extensions.

### CODE REVIEW

File: extensions/arc/src/extension.ts [hunk 1]
```diff
@@ -54,6 +50,19 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
 	});
 
 	await checkArcDeploymentExtension();
+
+	return {
+		getAzureArcTreeDataProvider: () => treeDataProvider
```
Comment (Charles-Gagnon) [inline extensions/arc/src/extension.ts:55]: Don't expose the whole tree provider - that's unnecessary and has weird consequences with other things being able to control our tree.

Just have a function to return the currently registered controllers instead.

File: extensions/arc/src/extension.ts [hunk 2]
```diff
@@ -54,6 +50,19 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
...
+	};
+}
+
+async function connectToController(treeDataProvider: AzureArcTreeDataProvider) {
```
Comment (Charles-Gagnon) [inline extensions/arc/src/extension.ts:59]: return type `Promise<void>`

Comment (Charles-Gagnon) [inline extensions/arc/src/extension.ts:59]: Why did you move this out? Just leave it in the command definition unless something else needs to use this.

**Review by Charles-Gagnon — CHANGES_REQUESTED** [review]
Hooking up the API looks fine but we shouldn't be returning the entire tree controller.

File: extensions/arc/src/extension.ts (hunk 1 above)
Comment (ranasaria) [reply extensions/arc/src/extension.ts:55]: so just an api to return treeDataProvider.getChildren()?

Comment (ranasaria) [reply extensions/arc/src/extension.ts:55]: and then later i will add an API to invoke the add/update controller dialog

File: extensions/arc/src/extension.ts (hunk 2 above)
Comment (ranasaria) [reply extensions/arc/src/extension.ts:59]: I was going to create a connectToController method as an API method but decided to postpone that to later. So I will revert this change for now.

File: extensions/arc/src/extension.ts (hunk 1 above)
Comment (Charles-Gagnon) [reply extensions/arc/src/extension.ts:55]: The dialog you can just use the existing command for - no need to add an api just for that.

And yes - the api here would be something like 

`getRegisteredDataControllers(): ControllerInfo[]` 

where ControllerInfo is https://github.com/microsoft/azuredatastudio/blob/main/extensions/arc/src/models/controllerModel.ts#L14 which is this : ControllerTreeNode.model.info

Comment (ranasaria) [reply extensions/arc/src/extension.ts:55]: done.

File: extensions/arc/src/extension.ts (hunk 2 above)
Comment (ranasaria) [reply extensions/arc/src/extension.ts:59]: reverted

Comment (ranasaria) [reply extensions/arc/src/extension.ts:59]: reverted

**Review by Charles-Gagnon — APPROVED** [review]
